src/main/java/emissary/place/Main.java

```diff
@@ -1040,33 +1042,71 @@ public void printPayload(IBaseDataObject payload) {
      * here. One payload or attachment per file. Directories will be created as needed underneath the baseOutputDir (can be
-     * specified with -d)
+     * specified with -d).
+     * <p>
+     * Note: Output will be suppressed if an IBDO form or filename would yield an output path outside the configured
+     * {@link #getBaseOutputDir() base output directory} path
+     * </p>
      *
      * @param payload the processed payload
      * @param att the list of attachments
      */
     public void handleSplitOutput(IBaseDataObject payload, List<IBaseDataObject> att) {
         String fn = getBaseOutputDir() + "/" + payload.shortName() + "." + payload.currentForm();
-        boolean status = Executrix.writeDataToFile(payload.data(), fn);
-        if (status) {
-            logger.debug("Wrote output to " + fn);
-        } else {
-            logger.error("Could not write output to " + fn);
-        }
 
+        String safePath;
+        try {
+            safePath = filePathIsWithinBaseDirectory(getBaseOutputDir(), fn);
+            if (Executrix.writeDataToFile(payload.data(), safePath)) {
+                logger.debug("Wrote output to {}", safePath);
+            } else {
+                logger.error("Could not write output to {}", safePath);
+            }
+        } catch (IllegalArgumentException e) {
+            logger.error("Could not write output to {}", fn, e);
+        }
         for (IBaseDataObject part : att) {
             fn = getBaseOutputDir() + "/" + part.shortName() + "." + part.currentForm();
-            status = Executrix.writeDataToFile(part.data(), fn);
-            if (status) {
-                logger.debug("Wrote attachment output to " + fn);
-            } else {
-                logger.error("Could not write output to " + fn);
+            try {
+                safePath = filePathIsWithinBaseDirectory(getBaseOutputDir(), fn);
+                if (Executrix.writeDataToFile(payload.data(), safePath)) {
+                    logger.debug("Wrote attachment output to " + safePath);
+                } else {
+                    logger.error("Could not write output to " + safePath);
+                }
+            } catch (IllegalArgumentException e) {
+                logger.error("Could not write output to {}", fn, e);
             }
         }
     }
 
+    /**
+     * Normalizes the provided paths and tests to see whether the file path is within the required base directory.
+     *
+     * @param requiredBase required base directory
+     * @param filePath file path to be tested
+     * @return the normalized file path is within the required base directory
+     * @throws IllegalArgumentException if the filePath contains illegal characters or is outside the required base
+     *         directory
+     */
+    static String filePathIsWithinBaseDirectory(String requiredBase, String filePath) throws IllegalArgumentException {
+        // probably an overly simplistic test
+        if (filePath.contains("..")) {
```

Comment:
The check on this line is a simple guard that satisfies a CodeQL [path sanitizer](https://github.com/github/codeql/blob/main/java/ql/src/Security/CWE/CWE-022/TaintedPath.ql#L22-L31) check, our sanization is considered (by CodeQL scans) sufficient to defeat [Uncontrolled data used in path expression](https://codeql.github.com/codeql-query-help/java/java-path-injection/)" vulnerability.

Lines 1099 to 1103 perform more thorough validation to ensure that we consider the `filePath` resolves to a location we trust (and doesn't try to perform path traversal to an unexpected location), but we need the guard on line 1096 so line 1100 isn't mistakenly flagged as vulnerable code. 

This change has no production impact:  no existing production code paths reach this method, but it's reachable by legacy `public static main(String[] args)` methods in numerous Place implementations that call the `ServiceProviderPlace.mainRunner` method for ad hoc testing.